src/backend/common/deferred/queues/rq_queue.py

```diff
@@ -29,6 +29,13 @@ def _task_request(
     def _enqueue(self, request: RQTaskRequest) -> None:
         import requests
 
+        # Add `X-Google-TBA-RedisTask` in dev to simulate auth mechanism
+        # If header is attempted to be added in prod to bypass auth, it will be
+        # dropped by App Engine.
+        # https://cloud.google.com/appengine/docs/standard/go/reference/request-response-headers#removed_headers
+        headers = request.headers
+        headers["X-Google-TBA-RedisTask"] = "true"
```

Comment:
- [ ] We should add some documentation around `X-Google-TBA-RedisTask` in the task queue stuff just to be like - this is what this is doing, this is why this is named this, etc.